### PR TITLE
Refactor slope_optim.

### PR DESCRIPTION
**Pull request by uyennguyen0903**
- Refactor thumbnailer_slope_optim.cc (remove the `new_optim_list` vector from `LossyEncodeSlopeOptim()` to avoid misunderstanding).
- Add flag `-equal_quality` to `main.cc`.

### CODE REVIEW

File: src/main.cc [hunk 1]
```diff
@@ -47,6 +47,8 @@ ABSL_FLAG(bool, allow_mixed, false, "Use mixed lossy/lossless compression.");
 ABSL_FLAG(bool, verbose, false, "Print various encoding statistics.");
 
 // Thumbnailer methods.
+ABSL_FLAG(bool, equal_quality, false,
+          "Generate animation so that all frames have the same quality.");
```
Comment (y-guyon) [inline src/main.cc]: What is the point of this flag if it is the default behavior?

**Review by y-guyon — APPROVED** [review]

File: src/main.cc (hunk 1 above)
Comment (uyennguyen0903) [reply src/main.cc]: I think it should be clearer to have this flag. Without this flag, if user intends to create animation with `equal_quality` method and write `-equal_quality` in command line, the thumbnailer will show error in this case.

Comment (y-guyon) [reply src/main.cc]: Then a single `ABSL_FLAG(std::string)` might make more sense than several `ABSL_FLAG(bool)` (see [comment](https://github.com/googleinterns/thumbnailer/pull/36#discussion_r488001003)), up to you

File: README.md [hunk 2]
```diff
@@ -51,29 +52,31 @@ The following example shows the format of the text file containing frame filenam
 ./anim1/frame04.png 600
 ```
 
-**WebP Encoding Options:**
+**Options:**
 
-| Option | Default Value | Description|
-|--------|:-------------:|------------|
+| <img width="150"> Option | Default Value | <img width="600"> Description|
```
Comment (y-guyon) [inline README.md]: Is there another way to center the text rather than manual padding with `` tags?

File: README.md [hunk 3]
```diff
@@ -51,29 +52,31 @@ The following example shows the format of the text file containing frame filenam
...
-|`-equal_quality`|true|Generate animation setting the same quality to all frames.|
-|`-equal_psnr`|false|Generate animation so that all frames have the same PSNR.|
-|`-near_ll_diff`|false|Generate animation allowing near-lossless method, impose different pre-processing to near-losslessly-encoded frames.|
-|`-near_ll_equal`|false|Generate animation allowing near-lossless method, impose the same pre-processing to near-losslessly-encoded frames.|
-|`-slope_optim`|false|Generate animation with slope optimization.|
+|`-algorithm`|equal_quality|Algorithm to generate animation {equal_quality, equal_psnr, near_ll_diff, near_ll_equal, slope_optim}.|
 |`-slope_dpsnr`|1.0|Maximum PSNR change (in dB) used in slope optimization.|
 |`-verbose`|false|Print various encoding statistics showed by chosen algorithm.|
 
+**`-algorithm` flag description:**
```
Comment (y-guyon) [inline README.md]: You should use `## Title` here I think. See https://google.github.io/styleguide/docguide/style.html

File: README.md [hunk 4]
```diff
@@ -51,29 +52,31 @@ The following example shows the format of the text file containing frame filenam
...
+
+| <img width="100"> Algorithm | <img width="600"> Description |
+|-----------------------------|-------------------------------|
+|`-equal_quality`|Generate animation setting the same quality to all frames.|
```
Comment (y-guyon) [inline README.md]: This is no longer `-equal_quality` but `-algorithm equal_quality`

File: src/main.cc [hunk 5]
```diff
@@ -140,15 +129,23 @@ int main(int argc, char* argv[]) {
...
-  } else if (absl::GetFlag(FLAGS_near_ll_equal)) {
+  } else if (method_flag == "near_ll_equal") {
+    // Generate animation allowing near-lossless method. Use the same
+    // pre-processing value for all near-lossless frames.
     method = libwebp::Thumbnailer::Method::kNearllEqual;
-  } else if (absl::GetFlag(FLAGS_slope_optim)) {
+  } else if (method_flag == "slope_optim") {
+    // Generate animation with slope optimization.
     method = libwebp::Thumbnailer::Method::kSlopeOptim;
   }
```
Comment (y-guyon) [inline src/main.cc:153]: `else { std::cerr << "Unknown -algorithm " << method_flag ...`

**Review by y-guyon — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (uyennguyen0903) [reply README.md]: I've tried several ways but it didn't work.

Comment (y-guyon) [reply README.md]: Up to you but it feels like a trick. If the purpose was to fix the broken flags split on two lines, just remove the `-` I guess

**Pull request merged**